doc/xplat_cli_migration.md

```diff
@@ -0,0 +1,205 @@
+  * Work with either ID values (`--ids`) _or_ Group and Name (`-g -n`)
+  * Built in client-side query engine powered by JMESPath
+* (Coming) Support for more features and services
+  * Our preview meets or exceeds XPlat CLI functionality for Compute, Storage, Network, RBAC, and ARM
+  * More services coming online soon!
+
+While we believe the above list is compelling, it's important to remember
+**the Azure CLI 2.0 only supports ARM mode**.  You must continue to use
+the Azure XPlat CLI for ASM/Classic mode management.
+
+## Getting both CLI's setup side-by-side
+
+First, you run `azure --version` and ensure you are using `0.10.5` or later, as
+this is required for sharing your credentials between both CLIs.  If you installed
+using NPM, upgrade with `npm upgrade -g azure-cli`.  If you used an installer,
+we recommend downloading the latest installer to upgrade.
+
+To install the Azure CLI 2.0, follow the steps for your preferred platform or
+environment on our [Installation Guide](https://github.com/Azure/azure-cli/blob/master/doc/preview_install_guide.md).
+
+Once installed, you can run `az configure` and follow the steps to setup the
+Azure CLI 2.0 as well as help login.  Once this step is complete you should be
+to use both CLIs.  
+
+## Important new concepts in the Azure CLI 2.0
+Here is a quick list of some new and changed concepts that can help you understand the new tool.
+
+* Interactive Concepts
+  * Use `az configure` to setup your default output format
+  * You will find help to be generally more useful, try `az vm create -h` for an example
+  * Positional parameters are not supported, use `az vm list -g MyGroup` instead of `azure vm list MyGroup`
+* Automation and Scripting Concepts
+  * You can refer to resources using their full ID URI with `--ids` or with the resource group and name using `-g [rg] -n [name]`
+  * Use `--query "[expression]"` to extract single values
+  * Use `--out tsv` when getting a single value to prevent any extra markup
+  * Use `@-` to pipe values such as `az vm list --query [0].[id] --out tsv | az vm show --ids @-`
+* Service Specific Concepts
+  * VM power state is no longer included in `az vm list`, use `az vm get-instance-view` instead
+
+## Moving scripts from XPlat CLI to Azure CLI 2.0
+
+Generally, converting a script from XPlat to Az follows these steps:
+
+1. Switch `azure` commands to `az` commands
+2. Update commands to use new input values
+3. Update script to use new output formats
+4. Use `--query` to extract values reliably
+
+Below, we break down each of these steps.
+
+### Finding and switching to `az` commands
+
+While most commands keep the same group and command names between the Azure XPlat CLI and the Azure CLI 2.0, we've built a [azure to az conversion table](https://github.com/Azure/azure-cli/blob/master/doc/azure2az_commands.rst) for common commands.
+
+Mutate operations now use the `update` verb instead of `set`.  
```

Comment:
It might be worth addressing generic update here since some commands in AZ lack the convenience arguments of Xplat and rely exclusively on the generic operations.